docs/proposals/groups.md

```diff
@@ -0,0 +1,94 @@
+```
+// ClusterGroup contains the set of users that belong to a given ClusterGroup.  The name may not contain ":"
+type ClusterGroup struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// Users is a list of Users that are members of the ClusterGroup
+	Users util.StringSet
+}
+```
+
+A reverse lookup index will be built in memory (similar to the project cache) to be allow the authenticator to make fast lookups for ClusterGroup membership determination.
+
+
+## OrgGroups
+OrgGroups look a lot like ClusterGroups, but they are always scoped to a particular Organization and they behave differently.  OrgGroups are managed by Organization.Admins (ClusterGroups are managed by cluster admins), OrgGroups can only contain Users that are Organization.Members (ClusterGroups can contain any user), OrgGroups can only be bound to roles in projects owned by the Organization (ClusterGroups can be bound to roles in any project).  They will be cluster scoped resources, but we'll indicate the Organization scope of a OrgGroup by prefixing every OrgGroup name with Organization name.
+```
+// OrgGroup contains the set of Users that belong to a given OrgGroup.  The name of a OrgGroup is required to 
+// to be in the form of "OrganizationName:OrgGroupName".
+type OrgGroup struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// Users is a list of Users that are members of the OrgGroup
+	Users util.StringSet
+}
+```
+
+A reverse lookup index will be built in memory (similar to the project cache) to be allow the authenticator to make fast lookups for OrgGroup membership determination.  A valid OrgGroup.Name can never be used as a ClusterGroup.Name and vice-versa.  This detail is important later on for authorization parity with kube and backwards compatibility with v3 of OpenShift.
+
+
+## Organizations
+An Organization is a cluster scoped API object that references Users and has subresources for accessing and modifying Organization scoped OrgGroups.
+```
+// Organization contains the set of Users that are members of the Organization.
+type Organization struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// AdminUsers is a list of Users that can administer this Organization.  By default a policy rule exists
+	// that allows all Users to inspect and modify OrgGroups of Organizations that they are an Admin for.
+	// Admins may NOT modify Organization membership.
+	AdminUsers util.StringSet
+	// AdminClusterGroups is a list of ClusterGroups that can administer this Organization.  By default a policy rule exists
+	// that allows all Users to inspect and modify OrgGroups of Organizations that they are an Admin for.
+	// Admins may NOT modify Organization membership.
+	AdminClusterGroups util.StringSet
+
+	// Members is a list of Users that are members of the Organization.  Only members may be assigned
+	// to OrgGroups.
+	Members util.StringSet
```

Comment:
Can ClusterGroups be members of OrgGroups? I would guess yes since they can already admin those.